src/xokdinst.rs

```diff
@@ -246,7 +257,37 @@ enum Opt {
     },
 }
 
-fn cmd_installer(version: Option<&str>) -> std::process::Command {
+/// Compute master memory: host memory - 6GB (4 for worker + 2 for bootstrap)
+fn get_master_mem(libvirt_auto_size: bool) -> u64 {
+    let default = 4096;
+    if !libvirt_auto_size {
+        return default;
+    }
+    match mem_info() {
+        Err(e) => {
+            eprintln!("{}: Couldn't get host total memory, using 4GB for master", e);
+            return default;
+        },
+        Ok(mem) => cmp::max(4096, (mem.avail - 6 * 1024 * 1024) / 1024),
+    }
+}
+
+/// Compute master cpu: host cpu - 2, minimum of 2
+fn get_master_cpu(libvirt_auto_size: bool) -> u32 {
+    let default = 2;
+    if !libvirt_auto_size {
+        return default;
+    }
+    match cpu_num() {
+        Err(e) => {
+            eprintln!("{}: Couldn't get host cpu count, use 2 cpu for master", e);
+            return default
+        },
+        Ok(cpu) => cmp::max(2, cpu - 2),
+    }
+}
+
+fn cmd_installer(version: Option<&str>, libvirt_auto_size: bool) -> std::process::Command {
```

Comment:
This is OK as is for now but we're probably going to need to clean this up in the future to have shared options between genconfig and launch, and pass that shared struct to this function.

Reply:
I'm not entirely sure how to do that, but that sounds like a good idea to remove the duplication from the Config struct :-)